tests/api/api/marketplace/test_marketplace.py

```diff
@@ -304,3 +304,25 @@ def test_marketplace_catalog_apis(
     function_modified_name = item["metadata"]["name"].replace("_", "-")
 
     assert yaml_function_name == function_modified_name
+
+
+def test_marketplace_get_asset(
+    db: Session, client: TestClient, k8s_secrets_mock: tests.api.conftest.K8sSecretsMock
+) -> None:
```

Comment:
That's a good test, but it's not really a unit-test (it's very similar to the `test_marketplace_default_source` test, only it works with http rather than directly with the crud) - it relies on the existing catalog. Can you add more unit-tests that look at more specific assets functionality -
* Verify assets are retrieved using source secrets
* Verify not-found assets are handled properly
* Ensure the actual content is reported properly (currently you just check the content type)